src/commands/roll.ts

```diff
@@ -0,0 +1,126 @@
+import { JeffreyGachaURLs, displayLegendary } from "../JeffreyGacha";
+import { SlashCommandBuilder, CommandInteraction, EmbedBuilder, InviteTargetType } from "discord.js";
+import { replyWithEmbed, rng } from '../utils';
+import {
+    addGacha,
+    addUserWalletsUser,
+    changeBalance,
+    checkBalance,
+    checkGachaInv,
+    checkUserWalletsUser,
+    gachaLvlUp,
+    checkGachaLevel
+} from "../DBUtils";
+
+export const Roll = {
+    info: new SlashCommandBuilder()
+        .setName('roll')
+        .setDescription('Roll for Jeffrey\'s!'),
+
+    run: async (interaction: CommandInteraction): Promise<void> => {
+        if (!interaction.guild) {
+            await interaction.reply('this command can only be done in a server');
+            return;
+        }
+        let embed;
+        let currentBalance;
+        const userID = interaction.user.id;
+        try {
```

Comment:
If your functions aren't throwing an error, then you shouldn't need to try catch